packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -284,6 +284,15 @@ class DriveCommand extends RunCommandBase {
       throwToolExit(null);
     }
     if (await _fileSystem.type(testFile) != FileSystemEntityType.file) {
+      // A very common source of error is holding "flutter drive" wrong,
+      // and providing the "test_driver/foo_test.dart" as the target, when
+      // the intention was to provide "lib/foo.dart".
+      if (targetFile.endsWith('_test.dart')) {
+        _logger.printError(
+          'The file path passed to --target should be an app entrypoint that '
+          'contains a "main()". Did you mean "flutter drive --driver $targetFile"?',
```

Comment:
Looking at the docs (https://docs.flutter.dev/testing/integration-tests), it looks like it's normal to have the driver file end with `_test.dart`. Am I missing something here?

Reply:
e.g.

```
flutter drive \
  --driver=test_driver/integration_test.dart \
  --target=integration_test/app_test.dart \
  -d chrome
  ```

Reply:
I guess we could do introspection that this message shows up if the file is in `test_driver`, which is what we are trying to catch.

Reply:
Done.